tests/Doctrine/Tests/DBAL/Query/Expression/CompositeExpressionTest.php

```diff
@@ -77,4 +77,12 @@ public function provideDataForConvertToString()
             ),
         );
     }
+
+    public function testAddPreventsAddingEmptyCompositeExpressionObjects()
+    {
+        $emptyExpr = new CompositeExpression(CompositeExpression::TYPE_AND, []);
+        $expr = new CompositeExpression(CompositeExpression::TYPE_AND, [$emptyExpr]);
+
+        $this->assertEquals(0, count($expr));
```

Comment:
I don't quite get it. The object is mutable, so you can add any more parts as you want as long as you do not cast it as `string`. Then you might want to check for number of parts >= 2 and eventually throw an exception. But your test is not revealing any issue. Also I am not sure if this is not considered part of the internal API /cc @Ocramius